pkg/controllers/proxy/network_services_controller.go

```diff
@@ -1085,6 +1092,223 @@ func (ln *linuxNetworking) prepareEndpointForDsr(containerID string, endpointIP
 	return nil
 }
 
+// The same as prepareEndpointForDsr but using CRI instead of docker
+func (ln *linuxNetworking) prepareEndpointForDsrWithCRI(runtimeEndpoint, containerID, endpointIP, vip string) error {
+
+	// FIXME: its possible switch namespaces may never work safely in GO without hacks.
+	//	 https://groups.google.com/forum/#!topic/golang-nuts/ss1gEOcehjk/discussion
+	//	 https://www.weave.works/blog/linux-namespaces-and-go-don-t-mix
+	// Dont know if same issue, but seen namespace issue, so adding
+	// logs and boilerplate code and verbose logs for diagnosis
+
+	if runtimeEndpoint == "" {
+		return errors.New("runtimeEndpoint is not specified")
+	}
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	var activeNetworkNamespaceHandle netns.NsHandle
+
+	hostNetworkNamespaceHandle, err := netns.Get()
+	if err != nil {
+		return errors.New("failed to get host namespace due to " + err.Error())
+	}
+	defer hostNetworkNamespaceHandle.Close()
+
+	activeNetworkNamespaceHandle, err = netns.Get()
```

Comment:
@ep4eg I know that you just copied this from the original function, but do you happen to know why it is necessary to try fetching the host namespace a second time early in the function as opposed to just fetching it once down below after you identify the containers network namespace?

Have you tested this implementation without this and seen it fail? It would be nice to remove un-needed logic if possible.

Reply:
Well, this is a complicated topic, as the entire method is kinda mess with a bunch of duplicated code that  I'd break into chunks tbh. Anywho, I didn't find any reason to keep this part, so I just removed that, tests passed, everything's working.